### PR TITLE
Add KOTLIN_SCRIPTS feature flag

### PR DESCRIPTION
**Pull request by jmfayard**
Minimum viable PR for #582

Discussion should happen in the issue

### CODE REVIEW

File: plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt [hunk 1]
```diff
@@ -59,4 +80,11 @@ open class RefreshVersionsCleanupTask : DefaultTask() {
             OutputFile.GRADLE_VERSIONS_CATALOG.logFileWasModified()
         }
     }
+
+    @TaskAction
+    fun cleanupKotlinScripts() {
+        if (FeatureFlag.KOTLIN_SCRIPTS.isEnabled) {
+            println("TODO: refreshVersionsCleanUp should clean up Kotlin Scripts see https://github.com/jmfayard/refreshVersions/issues/582")
```
Comment (LouisCAD) [inline plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt:87]:
```suggestion
            println("NOTE: refreshVersionsCleanUp doesn't clean up Kotlin Scripts yet, see https://github.com/jmfayard/refreshVersions/issues/582")
```

File: plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsTask.kt [hunk 2]
```diff
@@ -115,6 +115,9 @@ open class RefreshVersionsTask : DefaultTask() {
                 }
             }
         }
+        if (FeatureFlag.KOTLIN_SCRIPTS.isEnabled) {
+            println("TODO: refreshVersions should support Kotlin Scripts see https://github.com/jmfayard/refreshVersions/issues/582")
```
Comment (LouisCAD) [inline plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsTask.kt:119]:
```suggestion
            println("NOTE: refreshVersions support for Kotlin Scripts isn't implemented yet, see https://github.com/jmfayard/refreshVersions/issues/582")
```

File: plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt [hunk 3]
```diff
@@ -11,11 +11,32 @@ import de.fayard.refreshVersions.core.internal.versions.VersionsPropertiesModel.
...
+        }
+
+    @Input
+    @Optional
+    @Option(option = "disable", description = "Disable a feature flag")
+    var disableFlag: FeatureFlag? = null
+        set(value) {
+            field = value
+            if (value != null) FeatureFlag.userSettings.put(value, false)
+        }
```
Comment (LouisCAD) [inline plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt:38]: I think this should be in a separate PR, don't you?

**Review by LouisCAD — COMMENTED** [review]
A few inline comments.

Also, a general thought I had for a while: We probably want to not depend on being a Gradle plugin for Kotlin Scripts support.
That means potentially a lot of refactoring to make a Gradle independent core.
We can sure start with a version that depends on the Gradle plugin and a host Gradle project to run, but that's not something we want in the future I think.

If you want to iterate on that Kotlin scripts support soon, I suggest that we do a little brainstorming session on the phone, or IRL when I'll be in Paris on the 3rd of November, and I write an issue or a design doc about the considerations for a first version of the feature. That should help us avoid a long to review and fix PR like what happened for versions catalog support.

**Comment by jmfayard** [comment]
@LouisCAD I commented on your idea to have something independant from Gradle here https://github.com/jmfayard/refreshVersions/issues/586

I still think there is no issue with adding Kotlin Script support in the current codebase. 
We can refactor the code later.

Something I need to clarify is that I do **not** want to implement Kotlin Scripts support in one large pull request here like I did for Gradle Versions Catalogs. 
I learned my lesson that small is beautiful, and having multiple small pulll requests is less work and less stress for everyone that one huge pull request that does everything.
So this particular PR has the narrow goal of **adding the command-line flag and discussing how the feature should look like** but nothing more.

File: plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt (hunk 3 above)
Comment (jmfayard) [reply plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt:38]: actually no, see comment :)

**Comment by LouisCAD** [comment]
I don't think that getting this PR merged or not is an actual blocker to start working/iterating on the kotlin scripts support feature, but since we might want to add further feature flags at some point, which would generate little git conflicts, I guess it's a good idea to merge it now.
